schema2Markdown.js

```diff
@@ -0,0 +1,255 @@
+            markdownContent += `**Example Value**: \n`;
+            if (typeof example[propertyName] === 'object') {
+                markdownContent += `\`\`\`json\n${JSON.stringify(example[propertyName], null, 2)}\n\`\`\`\n\n`;
+            } else if (example[propertyName]) {
+                markdownContent += `\`${example[propertyName]}\`\n\n`;
+            }
+        });
+    }
+    
+    return markdownContent;
+}
+
+function renderType(ref) {
+    return `**Type**: \`${ref}\`\n\n`;
+}
+
+function renderEnum(ref) {
+    // for each item in ref, wrap it in backticks and join with a comma
+    return `**Possible values**: ${ref.map((item) => `\`${item}\``).join(', ')}\n\n`;
+}
+
+function renderRef(contextRef) {
+    const [filePath, objectPath] = contextRef.split('#'); // ../api/api.schema.json, /definitions/AppIdentifier
+    const objectType = filePath.split('/').pop().split('.')[0]; // api
+
+    // FROM ../api/api.schema.json#/definitions/AppIdentifier
+    // TO   ../api/schemas/Appidentifier
+
+    // FROM timerange.schema.json#
+    // TO   timerange/schemas/timerange
+
+    let objectName = objectType;
+    if (objectPath) {
+        objectName = objectPath.split('/').pop(); // AppIdentifier
+    }
+
+    let objectRef = objectName;
+    if (filePath.startsWith('../')) {
+        objectRef = "../../" + objectType + "/schemas/" + objectName;
+    }
+
+    return `**Reference**: [${objectType}/${objectName}](${objectRef})\n\n`;
+}
+
+function hasAllOf(allOfArray) {
+    return Array.isArray(allOfArray) && 
+        allOfArray.length > 0 && 
+        allOfArray[0] != null && 
+        allOfArray[0].properties != null
+}
+
+function hasProperties(schema) {
+    return schema.properties != null;
+}
+
+// Function to generate Markdown content from JSON schema
+function generateObjectMD(schema, title, schemaFolderName, filePath, version) {
+
+    const objectName = schema.title
+
+    if (schema.title != null) {
+        title = schema.title;
+    }
+
+    let markdownContent = `# ${title}\n\n`;
+
+    if (schema.description != null) {
+        markdownContent += `${escape(schema.description)}\n\n`; 
+    }
+
+    if (schema.type) {
+        markdownContent += renderType(schema.type);
+    }
+    if (schema.enum) {
+        markdownContent += renderEnum(schema.enum);
+    }
+
+    if (hasAllOf(schema.allOf) || hasProperties(schema)) {
+        // Extract properties, required fields, and $ref from the first allOf object
+        let root = schema;
+        if (hasAllOf(schema.allOf)) {
+            root = schema.allOf[0];
+        }
+        
+        const properties = root.properties;
+        const required = root.required;
+        const ref = root.$ref;
+
+        markdownContent += `## Properties\n\n`; 
+
+        for (const [propertyName, propertyDetails] of Object.entries(properties)) {
+            markdownContent += processProperty(propertyName, propertyDetails, schema.examples);
+        }
```

Comment:
Redner type at top level and skip under properties section:

```suggestion
        const properties = root.properties;
        const typeString = properties?.type?.const;
        const required = root.required;
        const ref = root.$ref;
                
        markdownContent += `## Type\n\n`;
        markdownContent += `\`${typeString}\`\n\n`;

        markdownContent += `## Properties\n\n`; 

        for (const [propertyName, propertyDetails] of Object.entries(properties)) {
            if (propertyName != "type"){ 
                markdownContent += processProperty(propertyName, propertyDetails, schema.examples);
            }
        }
```